### PR TITLE
Fix Job Tracking Issue

### PR DESCRIPTION
**Pull request by cgardens**
Issue: https://github.com/airbytehq/airbyte/issues/1064

## What
* Job metadata was not being sent properly to segment.
* Issue was that I had not set env variable properly in the scheduler.
* Added tracking for submission of a job and completion of a job.

### CODE REVIEW

File: airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java [hunk 1]
```diff
@@ -96,61 +98,86 @@ private void submitJob(Job job) {
...
+      }
     }
+    return metadata;
+  }
+
+  private static JobStatus getStatus(OutputAndStatus<?> output) {
+    return switch (output.getStatus()) {
+      case SUCCEEDED -> JobStatus.COMPLETED;
+      case FAILED -> JobStatus.FAILED;
+      default -> throw new IllegalStateException("Unknown state " + output.getStatus());
```
Comment (sherifnada) [inline airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java:179]: This should probably log instead of throw an exception

**Review by sherifnada — APPROVED** [review]

File: .env.dev [hunk 2]
```diff
@@ -13,7 +13,7 @@ WORKSPACE_ROOT=/tmp/workspace
 WORKSPACE_DOCKER_MOUNT=/tmp/dev_root/workspace
 LOCAL_ROOT=/tmp/airbyte_local
 LOCAL_DOCKER_MOUNT=/tmp/airbyte_local
-TRACKING_STRATEGY=logging
+TRACKING_STRATEGY=segment
```
Comment (cgardens) [inline .env.dev:16]: todo - revert before merge. going to do this separately.

File: airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java (hunk 1 above)
Comment (cgardens) [reply airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java:186]: why?

Comment (sherifnada) [reply airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java:186]: mb -- I misread. This is correct.

Comment (sherifnada) [reply airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java:186]: I thoguth this was only used in tracking code not in business logic. Wouldn't have wanted to fail over tracking issues.

File: airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java [hunk 3]
```diff
@@ -142,24 +142,21 @@ public void writeStandardDestinationDefinition(final StandardDestinationDefiniti
         destinationDefinition);
   }
 
-  public SourceConnection getSourceConnection(final UUID sourceId)
```
Comment (cgardens) [inline airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java:145]: blank space changes.

File: airbyte-server/src/main/java/io/airbyte/server/handlers/SchedulerHandler.java [hunk 4]
```diff
@@ -159,12 +159,6 @@ public SourceDefinitionSpecificationRead getSourceDefinitionSpecification(Source
     final String imageName = DockerUtils.getTaggedImageName(source.getDockerRepository(), source.getDockerImageTag());
     final ConnectorSpecification spec = getConnectorSpecification(imageName);
 
-    TrackingClientSingleton.get().track("get_source_spec", ImmutableMap.<String, Object>builder()
```
Comment (cgardens) [inline airbyte-server/src/main/java/io/airbyte/server/handlers/SchedulerHandler.java:162]: this is just throwing money in a bucket and lighting it on fire.

**Pull request merged**